packages/spacecat-shared-data-access/src/index.d.ts

```diff
@@ -534,6 +534,57 @@ export interface ImportUrl {
 
 }
 
+/**
+ * Represents an API Key entity.
+ */
+export interface ApiKey {
+    /**
+     * Retrieves the ID of the API Key.
+     */
+    getId: () => string;
+
+    /**
+     * Retrieves the hashed key value of the API Key.
+     */
+    getHashedKey: () => string;
```

Comment:
Thinking out loud: any reason that `hashedKey` couldn't be the partition key? Could avoid the need for a GSI.

Reply:
It could be the partition key. Then we'll need to move from the current model implementation. We won't inherit from Base anymore (so no default Id).